File: backend/src/main/java/ma/zs/alc/service/facade/collaborator/faq/FaqEtudiantCollaboratorService.java
package ma.zs.alc.service.facade.collaborator.faq;

import java.util.List;
import ma.zs.alc.bean.core.faq.FaqEtudiant;
import ma.zs.alc.dao.criteria.core.faq.FaqEtudiantCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface FaqEtudiantCollaboratorService extends  IService<FaqEtudiant,FaqEtudiantCriteria>  {

    List<FaqEtudiant> findByEtudiantId(Long id);
    int deleteByEtudiantId(Long id);
    long countByEtudiantId(Long id);
    List<FaqEtudiant> findByFaqTypeId(Long id);
    int deleteByFaqTypeId(Long id);
    long countByFaqTypeId(Long id);




}

File: frontend/src/app/shared/service/collaborator/faq/FaqEtudiantCollaborator.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {FaqEtudiantDto} from '../../../model/faq/FaqEtudiant.model';
import {FaqEtudiantCriteria} from '../../../criteria/faq/FaqEtudiantCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class FaqEtudiantCollaboratorService extends AbstractService<FaqEtudiantDto, FaqEtudiantCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'collaborator/faqEtudiant/';
    }

    public constrcutDto(): FaqEtudiantDto {
        return new FaqEtudiantDto();
    }

    public constrcutCriteria(): FaqEtudiantCriteria {
        return new FaqEtudiantCriteria();
    }
}
